app/models/user.rb

```diff
@@ -122,6 +122,9 @@ def moderatable
 
   after_save :nullify_authentication_id, if: :any_sso_authorizations?
 
+  after_destroy :update_account_first_admin, if: :account_first_admin?
+  after_create :promote_to_first_admin, if: proc { account && account.first_admin_id.nil? && admin? && !impersonation_admin? }
```

Comment:
looks like it's time for a method for `{ account && account.first_admin_id.nil? && admin? && !impersonation_admin? } ` it also works as documentation since reading the conditions does not give me the idea what is being checked.

Reply:
makes sense 👍